src/CommonITILObject.php

```diff
@@ -1451,8 +1451,11 @@ public function getActiveOrSolvedLastDaysForItem($itemtype, $items_id, $days)
                     [
                         'NOT' => [$this->getTable() . '.solvedate' => null],
                         new QueryExpression(
-                            "ADDDATE(" . $DB->quoteName($this->getTable()) .
-                            "." . $DB->quoteName('solvedate') . ", INTERVAL $days DAY) > NOW()"
+                            QueryFunction::dateAdd(
+                                date: static::getTable() . '.solvedate',
+                                interval: $days,
+                                interval_unit: 'DAY'
+                            ) . ' > ' . QueryFunction::now()
                         )
```

Comment:
We cannot have a query function as an array key. It is a QueryExpression and PHP doesn't allow non-int/string keys and doesn't automatically cast it. If it were cast to a string, it would still be treated as an identifier.

Reply:
Right, my bad.

Reply:
My first reaction was the same as @cedric-anne syntax feels bloated.
But to replace this kind date comparison with also dateadd, I don't have better to suggest without moving from arrays declaration (and it seems to much atm)